x-pack/plugins/apm/server/routes/traces/route.ts

```diff
@@ -52,6 +58,7 @@ const tracesRoute = createApmServerRoute({
       end,
     });
 
+    const seed = await getRandomSamplerSeed({ security, request });
```

Comment:
this can happen in parallel to `setupRequest`. It doesn't really matter in terms of performance, this should resolve practically immediately, but IMHO we can avoid the confusion from calling this separately by requesting it in parallel.